src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+    private Node<T> newNode;
+    private Node<T> temporaryNode;
+
+    private static class Node<T> {
+        protected T element;
+        protected Node<T> prev;
+        protected Node<T> next;
+
+        protected Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
+    public MyLinkedList() {
+        first = new Node<>(null, null, null);
+        last = new Node<>(null, null, null);
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            temporaryNode = new Node<>(first, value, last);
+            first = temporaryNode;
+            last = temporaryNode;
+            size++;
+            return;
+        }
+        newNode = new Node<>(temporaryNode, value, null);
+        temporaryNode.next = newNode;
+        temporaryNode = newNode;
+        last = temporaryNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index >= 0 && index < size) {
+            temporaryNode = indexToNode(index);
+            newNode = new Node<>(temporaryNode.prev, value, temporaryNode);
+            size++;
+            if (index == 0) {
+                first = newNode;
+                return;
+            }
+            temporaryNode.prev.next = newNode;
+            temporaryNode.prev = newNode;
+            return;
+        }
+        throw new IndexOutOfBoundsException("It is not next index!");
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return indexToNode(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        temporaryNode = indexToNode(index);
+        T oldRecord = temporaryNode.element;
+        temporaryNode.element = value;
+        return oldRecord;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return unLink(index);
     }
 
     @Override
     public boolean remove(T object) {
+        temporaryNode = first;
+        for (int i = 0; i < size; i++) {
+            if (temporaryNode.element == object || (temporaryNode.element != null
+                    && temporaryNode.element.equals(object))) {
+                remove(i);
+                return true;
+            }
```

Comment:
you can use indexToNode() to get the needed node and then just use unlink() on it